cmd/juju/application/updateseries.go

```diff
@@ -0,0 +1,187 @@
+machine.
+
+In the case of updating a machine's series, the --force option should be used
+with caution since using a charm on a machine running an unsupported series may
+cause unexpected behavior.  Alternately, if the requested series supported in
+later revisions of the charm, upgrade-charm can run beforehand.
+
+Examples:
+	juju update-series <application> <series>
+	juju update-series <application> <series> --force
+	juju update-series <machine> <series>
+	juju update-series <machine> <series> --force
+
+See also:
+    machines
+    status
+    upgrade-charm
+`
+
+func (c *updateSeriesCommand) Info() *cmd.Info {
+	return &cmd.Info{
+		Name:    "update-series",
+		Args:    "[<application>|<machine>] <series>",
+		Purpose: "Update an application or machine's series.",
+		Doc:     updateSeriesDoc,
+	}
+}
+
+func (c *updateSeriesCommand) SetFlags(f *gnuflag.FlagSet) {
+	c.ModelCommandBase.SetFlags(f)
+	f.BoolVar(&c.force, "force", false, "Update even if the series is not supported by the charm and/or related subordinate charms.")
+}
+
+// Init implements cmd.Command.
+func (c *updateSeriesCommand) Init(args []string) error {
+	switch len(args) {
+	case 2:
+		if names.IsValidApplication(args[0]) {
+			c.applicationName = args[0]
+		} else if names.IsValidMachine(args[0]) {
+			c.machineNumber = args[0]
+		} else {
+			return errors.Errorf("invalid application or machine name %q", args[0])
+		}
+		c.series = args[1]
+	case 1:
+		if names.IsValidMachine(args[0]) {
+			return errors.Errorf("no series specified")
+		}
+		return errors.Errorf("no application name or no series specified")
+	case 0:
+		return errors.Errorf("no arguments specified")
+	default:
+		return cmd.CheckEmpty(args[2:])
+	}
+	return nil
+}
+
+// Run implements cmd.Run.
+func (c *updateSeriesCommand) Run(ctx *cmd.Context) error {
+
+	if c.updateSeriesClient == nil && c.charmClient == nil {
+		apiRoot, err := c.NewAPIRoot()
+		if err != nil {
+			return errors.Trace(err)
+		}
+		c.apiRoot = apiRoot
+		c.updateSeriesClient = application.NewClient(c.apiRoot)
+		c.charmClient = charms.NewClient(c.apiRoot)
+	}
+
+	if c.apiRoot.BestFacadeVersion("Application") < 5 {
+		suffix := "this server"
+		if version, ok := c.apiRoot.ServerVersion(); ok {
+			suffix = fmt.Sprintf("server version %s", version)
+		}
+		return errors.New("updating the application series is not supported by " + suffix)
+	}
+
+	if c.machineNumber != "" {
+		// TODO(hml) 2017/08/14
+		// implment updateMachineSeries
+		return errors.NotSupportedf("update-series by machine name is")
+	}
+
+	return c.updateApplicationSeries(ctx)
+}
+
+func (c *updateSeriesCommand) updateApplicationSeries(ctx *cmd.Context) error {
+
+	applicationInfo, err := c.updateSeriesClient.Get(c.applicationName)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	deployedSeries := applicationInfo.Series
+	if c.series == deployedSeries {
+		return errors.New(fmt.Sprintf("series %s is already in use by %s", c.series, c.applicationName))
```

Comment:
With this check here - currently there's a TODO in the apiserver to where the series is the same as what's been asked for. I don't think we need this check here - just go with it being a no-op and let the server side return nil.